chrome/content/zotfile/zotfile.js

```diff
@@ -455,16 +455,15 @@ Zotero.ZotFile = new function() {
      */
     this.formatSubfolder = function(item, format) {
         if (format == '') return '';
-        if(format.slice(-1) == this.folderSep) format = format.slice(0, -1);
+        if (format.slice(-1) == this.folderSep) format = format.slice(0, -1);
+        if (format[0] == this.folderSep) format = format.slice(1);
         var subfolder = this.Wildcards.replaceWildcard(item, format);
-        if (subfolder[0] == this.folderSep) subfolder = subfolder.slice(1);
-        // replace invalid characters        
+        // replace invalid characters
         subfolder = OS.Path.split(subfolder).components
             .map(s => s == '' ? 'undefined' : s)
+            .filter(s => s != this.Wildcards.emptyCollectionPlaceholder)
```

Comment:
It should be filtered out instead of substituted by empty string because `Zotero.File.getValidFileName` will modify empty components to `_`. So I'm using `filter`.